consensus/consensus-types/src/timeout_2chain.rs

```diff
@@ -133,24 +135,36 @@ impl TwoChainTimeoutCertificate {
         // Verify the highest timeout validity.
         self.timeout.verify(validators)?;
         let hqc_round = self.timeout.hqc_round();
-        let mut signed_round = 0;
-        validators.check_voting_power(self.signatures.keys())?;
-        for (author, (qc_round, signature)) in &self.signatures {
-            let t = TimeoutSigningRepr {
+        let timeout_messages: Vec<_> = self
+            .signatures_with_rounds
+            .get_voters_and_rounds(
+                &validators
+                    .get_ordered_account_addresses_iter()
+                    .collect_vec(),
+            )
+            .into_iter()
+            .map(|(_, round)| TimeoutSigningRepr {
                 epoch: self.timeout.epoch(),
                 round: self.timeout.round(),
-                hqc_round: *qc_round,
-            };
-            validators
-                .verify(*author, &t, signature)
-                .with_context(|| format!("Failed to verify {}'s TimeoutSigningRepr", *author))?;
-            signed_round = std::cmp::max(signed_round, *qc_round);
-        }
+                hqc_round: round,
+            })
+            .collect();
+        let timeout_messages_ref: Vec<_> = timeout_messages.iter().collect();
+        validators.verify_aggregated_signatures(
+            &timeout_messages_ref,
+            self.signatures_with_rounds.aggregated_sig(),
+        )?;
+        let signed_hqc = self
+            .signatures_with_rounds
+            .rounds()
+            .iter()
+            .max()
+            .expect("Empty rounds");
```

Comment:
I'd prefer to return error or unwrap_or(0) instead of expect even though it's probably safe since it passes the sig verification

Reply:
Since this passes the signature check, I think panicking here is apt as otherwise, something really bad has happened. Would prefer to leave it as is.